OpenQA/Isotovideo/dewebsockify.pm

```diff
@@ -0,0 +1,152 @@
+                                $log->trace("WebSocket text message: $text");
+                        });
+                        $tx->on(binary => sub ($tx, $bytes) {
+                                $log->trace("WebSocket binary message:\n" . sprintf("%v02X", $bytes));
+                                $stream->write($bytes) if $stream;
+                        });
+
+                        # pass pending data from raw socket to websocket
+                        $tx->send($_) for @tosend;
+                        @tosend = ();
+
+                        # handle websocket connection finish
+                        # note: Terminating here because at least for VMWare one needed a new URL/ticket anyways.
+                        $tx->on(finish => sub ($tx, $code, $reason) {
+                                $log->info("WebSocket closed with status $code.");
+                                $ws_connection = undef;
+                                $stream->close_gracefully if $stream;
+                                Mojo::IOLoop->stop_gracefully;
+                        });
+                });
+            }
+
+            # pass data from raw socket to websocket
+            $stream->on(read => sub ($s, $bytes) {
+                    if ($ws_connection) {
+                        $log->debug("Raw socket message:\n" . sprintf("%v02X", $bytes));    # uncoverable statement
+                        $ws_connection->send({binary => $bytes});    # uncoverable statement
+                    }
+                    else {
+                        $log->debug("Raw socket message (forwarding later):\n" . sprintf("%v02X", $bytes));    # uncoverable statement
+                        push @tosend, $bytes;    # uncoverable statement
+                    }
+            });
+
+            # handle raw socket close/error
+            $stream->on(close => sub ($s) {
+                    $log->info('Client closed connection');
+                    $stream = undef;
+                    $ws_connection ? $ws_connection->finish : Mojo::IOLoop->stop_gracefully;
+            });
+            $stream->on(error => sub ($stream, $err) {
+                    $log->error("Client error: $err");    # uncoverable statement
+            });
+
+            $log->info('Client accepted');
+    });
+    $server->listen(port => $port);
+    $server->start;
+    Mojo::IOLoop->start unless Mojo::IOLoop->is_running;
+}
+
+# Entry point for script execution
+sub parse_args {
+    my %options;
+    GetOptions(\%options, 'websocketurl=s', 'listenport=s', 'cookie=s', 'loglevel=s', 'insecure', 'help|h|?')
+      or usage(1);
+    usage(0) if $options{help};
+    return \%options;
+}
+
+sub usage ($r = 0) {
+    say 'Listens on a TCP port forwarding data to the specified websocket server
+    example: --websocketurl wss://... --listenport 590x --cookie "vmware_client=VMware; some_session=foobar" --insecure';
+    exit $r;
+}
+
+main(parse_args()) if not caller();
```

Comment:
This should be in the script itself.
The way it is now is kind of hidden, and also it doesn't work because caller will always return something true.